Formula/lib/libbsd.rb

```diff
@@ -26,6 +25,6 @@ def install
   end
 
   test do
-    assert_match "strtonum", shell_output("nm #{lib/"libbsd.so.#{version}"}")
+    assert_match "strtonum", shell_output("nm #{lib/shared_library("libbsd")}")
```

Comment:
Hmm it seems that `libbsd.so` on Linux is a linker script so `nm` doesn't work.

The following should work since we have `libbsd.0.dylib` on macOS and `libbsd.so.0` on Linux (and neither is a linker script).

```suggestion
    libbsd = lib/shared_library("libbsd", version.major.to_s)
    assert_match "strtonum", shell_output("nm #{libbsd}")
```

Also, can you please squash all the commits (including the suggested change here) into one with the message `libbsd: remove dependency on 'linux'` following [our commit style](https://docs.brew.sh/Formula-Cookbook#commit)? Thanks!

Reply:
> The following should work since we have `libbsd.0.dylib` on macOS and `libbsd.so.0` on Linux (and neither is a linker script).

Thank you. I'll try it.

> Also, can you please squash all the commits (including the suggested change here) into one with the message `libbsd: remove dependency on 'linux'` following [our commit style](https://docs.brew.sh/Formula-Cookbook#commit)? Thanks!

Once it is working I will definitely do that, thanks!

Reply:
@ZhongRuoyu @Bo98 It looks like CI tests finished successfully this time, so I have now squashed into one commit.
I think I followed the guidelines properly. Please let me know if I made a mistake. Thanks for all your help!